### PR TITLE
[theiacov] Add additional vadr output files & tarball; upgrade VADR docker

### LINKED ISSUE #464: output VADR annotations (esp. for Flu) & upgrade to VADR docker w/ new Flu model files
:cool:

### :pushpin: Explain the Request

VADR outputs lots of files and it would be helpful to output the annotation file, especially in the context of Influenza. I think we want to output the .sqa file, but perhaps others? (I'm not super familiar with all output files)

```
# Output printed to screen saved in:                              <samplename>.irma.consensus.vadr.log
# List of executed commands saved in:                             <samplename>.irma.consensus.vadr.cmd
# List and description of all output files saved in:              <samplename>.irma.consensus.vadr.filelist
# esl-seqstat -a output for input fasta file saved in:            <samplename>.irma.consensus.vadr.seqstat
# 5 column feature table output for passing sequences saved in:   <samplename>.irma.consensus.vadr.pass.tbl
# 5 column feature table output for failing sequences saved in:   <samplename>.irma.consensus.vadr.fail.tbl
# list of passing sequences saved in:                             <samplename>.irma.consensus.vadr.pass.list
# list of failing sequences saved in:                             <samplename>.irma.consensus.vadr.fail.list
# list of alerts in the feature tables saved in:                  <samplename>.irma.consensus.vadr.alt.list
# fasta file with passing sequences saved in:                     <samplename>.irma.consensus.vadr.pass.fa
# fasta file with failing sequences saved in:                     <samplename>.irma.consensus.vadr.fail.fa
# per-sequence tabular annotation summary file saved in:          <samplename>.irma.consensus.vadr.sqa
# per-sequence tabular classification summary file saved in:      <samplename>.irma.consensus.vadr.sqc
# per-feature tabular summary file saved in:                      <samplename>.irma.consensus.vadr.ftr
# per-model-segment tabular summary file saved in:                <samplename>.irma.consensus.vadr.sgm
# per-alert tabular summary file saved in:                        <samplename>.irma.consensus.vadr.alt
# alert count tabular summary file saved in:                      <samplename>.irma.consensus.vadr.alc
# per-model tabular summary file saved in:                        <samplename>.irma.consensus.vadr.mdl
# alignment doctoring tabular summary file saved in:              <samplename>.irma.consensus.vadr.dcr
```

~~I'm planning to upgrade the default VADR docker image with a PR related to IRMA and Flu changes, but will not update the VADR output parsing just yet.~~ Nevermind, the docker image upgrade and this request should be done in a single PR. I'll keep the IRMA stuff separate

The new docker image has the latest version (v1.6.3-2) of the Flu VADR models: https://github.com/StaPH-B/docker-builds/tree/master/vadr/1.6.3-hav-flu2

### PR DESCRIPTION
**Pull request by kapsakcj**
This PR closes #464

🗑️ This dev branch should be deleted after merging to main.

## :brain: Summary

This PR updates the default VADR docker image to one that includes the most up-to-date VADR model files for Influenza `v1.6.3-2` (previous docker image included influenza models `v1.6.3-1`). Changes are described here: https://bitbucket.org/nawrockie/vadr-models-flu/src/master/00RELEASE-NOTES.txt This model version will likely be used by NCBI for screening Flu genomes submitted to GenBank in the future, when they stop using FLAN.

A more in-depth explanation of the creation of the VADR model files for Influenza can is described in this pre-print (will likely be published soon): https://www.biorxiv.org/content/10.1101/2024.03.21.585980v1

VADR docs on Flu usage: https://github.com/ncbi/vadr/wiki/Influenza-annotation#example-annotation-of-flu-sequences-1

This PR also adds additonal outputs from VADR that were not previously exposed to the user. This will help in the context of Flu and other organisms when preparing genomes for submission to NCBI. 

The feature_tbl files produced by VADR may be useful (and maybe required?) when submitting Flu genomes to FASTA as they are compatible with BankIt. https://www.ncbi.nlm.nih.gov/WebSub/html/help/feature-table.html

## :zap: Impacted Workflows/Tasks

This impacts all workflows that call the VADR task:

- theiacov_illumina_pe
- theiacov_illumina_se
- theiacov_ONT
- theiacov_clearlabs
- theiacov_FASTA
- VADR_Update

**Note: VADR is run on multiple different organisms, so this impacts all organisms**

This PR may lead to different results in pre-existing outputs: **Extremely rare & unlikely, but Flu B samples may be annotated differently due to updated VADR models for influenza**

This PR uses an element that could cause duplicate runs to have different results: **No**

### :gear: Algorithm

- default docker image has been upgraded. Still using v1.6.3 of VADR, but the difference is that Flu model files have been upgraded within the docker image. Details described above.

### ➡️ Inputs

N/A

### ⬅️ Outputs

- `File? vadr_feature_tbl_pass` - 5 column feature table output for passing sequences. This output was not previously exposed at the workflow level but is captured at the task level.
- `File? vadr_feature_tbl_fail` - 5 column feature table output for failing sequences. This file will be helpful for identifying why a specific genomic feature was rejected or given a warning by VADR.
- `File? vadr_classification_summary_file` - per-sequence tabular classification summary file
- `File? vadr_all_outputs_tar_gz` - a tar.gz file containing all of VADR's output files in case users want to investigate a sample deeply. This output was not previously exposed at the workflow level, but is captured at the task level

## :test_tube: Testing

- sars-cov-2 samples via theiacov_FASTA_PHB: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/80ccb33f-bc32-4d7f-9a80-9b19c2806667
  - ~~⚠️ need to review outputs~~ Outputs look good ✅ 
  - `.fail.tbl` files have content for samples with VADR alerts, as expected
  - `.pass.tbl` files have content for samples with 0 VADR alerts, as expected.
  - All `.sqc.txt` files have content regardless of pass or fail.
  - The tarball `vadr_all_outputs_tar_gz` has all VADR output files, as expected. Was able to extract & view files within the tarball on Windows 11 computer without issue.
- Mpox samples via theiacov_Illumina_PE_PHB: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/3da52157-776c-4716-bc24-6e5b3e239f89
  - ~~⚠️ need to review outputs~~ Outputs look good ✅ 
  - `.fail.tbl` files have content for samples with VADR alerts, as expected
  - `.pass.tbl` files have content for samples with 0 VADR alerts, as expected.
  - All `.sqc.txt` files have content regardless of pass or fail.
  - The tarball `vadr_all_outputs_tar_gz` has all VADR output files, as expected. Was able to extract & view files within the tarball on Windows 11 computer without issue.
- Flu samples via theiacov_illumina_PE_PHB: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/bbfc5ae6-8741-434b-8cd3-c75f319569e1
  - ~~⚠️ need to review outputs~~ Outputs look good ✅ . Ignore the 1 failure, the original input FASTQ files are corrupted, so I expected failure.
  - each genome segment is evaluated separately, so for most samples, the `.fail.tbl` files and the `.pass.tbl` files have content depending on VADR passing or failing the segment
  - `.fail.tbl` files have content, as expected
  - `.pass.tbl` files have content, as expected.
  - The tarball `vadr_all_outputs_tar_gz` has all VADR output files, as expected. Was able to extract & view files within the tarball on Windows 11 computer without issue.
  - All `.sqc.txt` files have content regardless of pass or fail. Here's an example `.sqc.txt` file for an H1N1 sample that had passed & failed segments:

```
#seq  seq                  seq                                  sub                    seq    mdl         num                            sub   score  diff/  seq   
#idx  name                 len  p/f   ant  model1    grp1       grp1   score  sc/nt    cov    cov  bias  hits  str  model2    grp2       grp2   diff     nt  alerts
#---  ------------------  ----  ----  ---  --------  ---------  ----  ------  -----  -----  -----  ----  ----  ---  --------  ---------  ----  -----  -----  ------
1     A1_H1N1_PC_A_HA_H1  1701  PASS  yes  CY000449  fluA-seg4  H1    1094.7  0.644  0.999  0.956  39.9     1    +  DQ864721  fluA-seg4  H5    437.7  0.257  -     
2     A1_H1N1_PC_A_MP      982  PASS  yes  CY002009  fluA-seg7  -      778.8  0.793  1.000  0.956   0.5     1    +  CY103887  fluA-seg7  -     354.4  0.361  -     
3     A1_H1N1_PC_A_NA_N1  1410  PASS  yes  CY002538  fluA-seg6  N1     957.2  0.679  0.999  0.965  21.3     1    +  CY005359  fluA-seg6  N4    435.4  0.309  -     
4     A1_H1N1_PC_A_NP     1497  PASS  yes  CY006079  fluA-seg5  -     1147.8  0.767  0.999  0.955  23.4     1    +  CY125946  fluA-seg5  -     424.7  0.284  -     
5     A1_H1N1_PC_A_NS      863  PASS  yes  CY002284  fluA-seg8  -      623.2  0.722  0.999  0.969   6.8     1    +  CY103888  fluA-seg8  -     405.0  0.469  -     
6     A1_H1N1_PC_A_PA     1648  FAIL  yes  CY003645  fluA-seg3  -     1194.7  0.725  0.998  0.736  28.7     2    +  CY125944  fluA-seg3  -     417.8  0.254  -     
7     A1_H1N1_PC_A_PB1    2274  PASS  yes  CY003646  fluA-seg2  -     2153.0  0.947  1.000  0.971  60.1     1    +  CY125943  fluA-seg2  -     975.7  0.429  -     
8     A1_H1N1_PC_A_PB2    2024  FAIL  yes  CY002079  fluA-seg1  -     1467.1  0.725  0.999  0.868  29.6     3    +  CY103881  fluA-seg1  -     718.7  0.355  -     
```

### Suggested Scenarios for Reviewer to Test

Would be good to test a variety of organisms that are applicable to VADR across a variety of impacted workflows

## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [x] All changed results have been confirmed
- [x] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [x] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments
- [x] The documentation has been updated

### CODE REVIEW

File: tasks/quality_control/advanced_metrics/task_vadr.wdl
Comment (kapsakcj) [inline tasks/quality_control/advanced_metrics/task_vadr.wdl:50]: miniwdl linter said this was a "useless cat", so I adjusted it 😆

File: tests/workflows/theiacov/test_wf_theiacov_illumina_pe.yml
Comment (kapsakcj) [inline tests/workflows/theiacov/test_wf_theiacov_illumina_pe.yml:140]: FYI some of these CI updates in theiacov_illumina_PE wf and theiacov_clearlabs wf were required as they were missed over on PR #527 . ILMN PE and ClearLabs tests were not triggered in the CI and missed on accident.

@cimendes @jrotieno hopefully these CI changes are not unexpected? Can you check to make sure it's plausible & expected for these hashes & string searches to be changed?

Comment (cimendes) [reply tests/workflows/theiacov/test_wf_theiacov_illumina_pe.yml:140]: This is an expected change as now reads are interleaved prior to scrubbing, so the number of spots removed includes information for all reads. This was done to ensure that if a read is removed, so is its mate. @kapsakcj if you want to still verify this behaviour in the CI, you can change it to `miniwdl_run/call-read_QC_trim/call-ncbi_scrub_pe/work/SPOTS_REMOVED` and verify the md5sum of that.

**Comment by kapsakcj** [comment]
I'm going to mark ready for review, as the code changes have been tested successfully in Terra (for most workflows). I'm confident that the workflows not tested will run successfully as we are simply exposing a few new output files.

I am working on documentation and will reply back (and check the box) when that part is completed. Please do not merge until that is done.

**Comment by kapsakcj** [comment]
I've updated the TheiaCoV outputs table in the documentation.

No diagram updates are necessary for this.

OK to merge after your review is complete 👍

**Review by sage-wright — COMMENTED** [review]
code changes look good; testing [flu ont here](https://app.terra.bio/#workspaces/theiagen-validations/sage_viral/job_history/586eaf67-fd35-4d94-b787-9d4a0b23f642) and [multiple orgs fasta here](https://app.terra.bio/#workspaces/theiagen-validations/sage_viral/job_history/d043550f-4afd-4223-8af6-d0c68571a086) and will merge upon appropriate results

**Review by sage-wright — APPROVED** [review]
results are as expected; well done!

**Pull request merged**
